client/hostvolumemanager/host_volume_plugin.go

```diff
@@ -110,20 +124,65 @@ func (p *HostVolumePluginMkdir) Create(_ context.Context,
 		return resp, nil
 	} else if !os.IsNotExist(err) {
 		// doesn't exist, but some other path error
-		log.Debug("error with plugin", "error", err)
+		log.Debug("error with path", "error", err)
 		return nil, err
 	}
 
-	err := os.MkdirAll(path, 0o700)
+	params, err := decodeMkdirParams(req.Parameters)
 	if err != nil {
-		log.Debug("error with plugin", "error", err)
+		log.Error("error with parameters", "error", err)
 		return nil, err
 	}
 
+	err = os.MkdirAll(path, params.Mode)
+	if err != nil {
+		log.Error("error creating directory", "error", err)
+		return nil, fmt.Errorf("error creating directory: %w", err)
+	}
+
+	// Chown note: A uid or gid of -1 means to not change that value.
```

Comment:
I would say that this is a wild API decision from the stdlib, but it's actually a wild API decision from _Unix_. :grin: